sources/packages/api/src/services/restriction/student-restriction.service.ts

```diff
@@ -268,4 +223,40 @@ export class StudentRestrictionService extends RecordDataModelService<StudentRes
     } as Note;
     return this.repo.save(studentRestrictionEntity);
   }
+  /**
+   * The service function checks if the requested student has
+   * the restricted restriction.
+   * @param studentId student Id
+   * @param restrictionActions list of restriction actions
+   * @param checkAll,is by default false, checkAll decides if the all the
+   * elements in the restrictionActions should be checked (i.e checkAll = true)
+   * or any one of the element is to be checked (i.e checkAll = false).
+   * @returns boolean, true if the restriction action is present
+   * for the student, else false.
+   */
+  async isRestrictionActionExists(
+    studentId: number,
+    restrictionActions: RestrictionActionType[],
+    checkAll = false,
+  ): Promise<boolean> {
+    const query = this.repo
+      .createQueryBuilder("studentRestrictions")
+      .select("1")
+      .innerJoin("studentRestrictions.restriction", "restrictions")
+      .innerJoin("studentRestrictions.student", "student")
+      .where("studentRestrictions.isActive = true")
+      .andWhere("student.id= :studentId", {
+        studentId,
+      });
+    if (checkAll) {
+      query.andWhere("restrictions.actionType @> :restrictionActions", {
+        restrictionActions,
+      });
+    } else {
+      query.andWhere("restrictions.actionType && :restrictionActions", {
+        restrictionActions,
+      });
+    }
+    return (await query.getRawMany()).length > 0;
```

Comment:
to check if atleast one exist, we can use 

`const restriction= query.limit(1).getRawOne();
return !!restriction`